<launch>

<!-- test 3rd generation trackedrobot stair finding function -->



<!-- threshold t1 is the depth value,if it is bigger than t1 , then it is a obstacle point. -->
<arg name="Obstacle_NameSpace" default="ob3tacle_avoid" />
<arg name="USM_NameSpace" default="Up_Stair_Mode" />
<arg name="option" default=""/>

<!-- <rosparam param="xr_list">[0.0000,0.1000,0.2000,0.3000,0.4000,0.5000,0.6000,0.7000,0.8000,0.9000,1.0000,1.1000,1.2000,1.3000,1.4000,1.5000,1.6000,1.7000,1.8000,1.9000,2.0000,2.1000,2.2000,2.3000,2.4000,2.5000,2.6000,2.7000,2.8000,2.9000,3.0000,3.1000,3.2000,3.3000,3.4000,3.5000,3.6000,3.7000,3.8000,3.9000,4.0000,4.1000,4.2000,4.3000,4.4000,4.5000,4.6000,4.7000,4.8000,4.9000,5.0000,5.1000,5.2000,5.3000,5.4000,5.5000,5.6000,5.7000,5.8000,5.9000,6.0000,6.1000,6.2000,6.3000,6.4000,6.5000,6.6000,6.7000,6.8000,6.9000,7.0000,7.1000,7.2000,7.3000,7.4000,7.5000,7.6000,7.7000,7.8000,7.9000,8.0000,8.1000,8.2000,8.3000,8.4000,8.5000,8.6000,8.7000,8.8000,8.9000,9.0000,9.1000,9.2000,9.3000,9.4000,9.5000,9.6000,9.7000,9.8000,9.9000,10.0000,10.1000,10.2000,10.3000,10.4000,10.5000,10.6000,10.7000,10.8000,10.9000,11.0000,11.1000,11.2000,11.3000,11.4000,11.5000,11.6000,11.7000,11.8000,11.9000,12.0000,12.1000,12.2000,12.3000,12.4000,12.5000,12.6000,12.7000,12.8000,12.9000,13.0000,13.1000,13.2000,13.3000,13.4000,13.5000,13.6000,13.7000,13.8000,13.9000,14.0000,14.1000,14.2000,14.3000,14.4000,14.5000,14.6000,14.7000,14.8000,14.9000,15.0000,15.1000,15.2000,15.3000,15.4000,15.5000,15.6000,15.7000,15.8000,15.9000,16.0000,16.1000,16.2000,16.3000,16.4000,16.5000,16.6000,16.7000,16.8000,16.9000,17.0000,17.1000,17.2000,17.3000,17.4000,17.5000,17.6000,17.7000,17.8000,17.9000,18.0000,18.1000,18.2000,18.3000,18.4000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.5000,18.6000,18.7000,18.8000,18.9000,19.0000,19.1000,19.2000,19.3000,19.4000,19.5000,19.6000,19.7000,19.8000,19.9000,20.0000,20.1000,20.2000,20.3000,20.4000,20.5000,20.6000,20.7000,20.8000,20.9000,21.0000,21.1000,21.2000]</rosparam>
<rosparam param="yr_list">[0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,-0.1000,-0.2000,-0.3000,-0.4000,-0.5000,-0.6000,-0.7000,-0.8000,-0.9000,-1.0000,-1.1000,-1.2000,-1.3000,-1.4000,-1.5000,-1.6000,-1.7000,-1.8000,-1.9000,-2.0000,-2.1000,-2.2000,-2.3000,-2.4000,-2.5000,-2.6000,-2.7000,-2.8000,-2.9000,-3.0000,-3.1000,-3.2000,-3.3000,-3.4000,-3.5000,-3.6000,-3.7000,-3.8000,-3.9000,-4.0000,-4.1000,-4.2000,-4.3000,-4.4000,-4.5000,-4.6000,-4.7000,-4.8000,-4.9000,-5.0000,-5.1000,-5.2000,-5.3000,-5.4000,-5.5000,-5.6000,-5.7000,-5.8000,-5.9000,-6.0000,-6.1000,-6.2000,-6.3000,-6.4000,-6.5000,-6.6000,-6.7000,-6.8000,-6.9000,-7.0000,-7.1000,-7.2000,-7.3000,-7.4000,-7.5000,-7.6000,-7.7000,-7.8000,-7.9000,-8.0000,-8.1000,-8.2000,-8.3000,-8.4000,-8.5000,-8.6000,-8.7000,-8.8000,-8.9000,-9.0000,-9.1000,-9.2000,-9.3000,-9.4000,-9.5000,-9.6000,-9.7000,-9.8000,-9.9000,-10.0000,-10.1000,-10.2000,-10.3000,-10.4000,-10.5000,-10.6000,-10.7000,-10.8000,-10.9000,-11.0000,-11.1000,-11.2000,-11.3000,-11.4000,-11.5000,-11.6000,-11.7000,-11.8000,-11.9000,-12.0000,-12.1000,-12.2000,-12.3000,-12.4000,-12.5000,-12.6000,-12.7000,-12.8000,-12.9000,-13.0000,-13.1000,-13.2000,-13.3000,-13.4000,-13.5000,-13.6000,-13.7000,-13.8000,-13.9000,-14.0000,-14.1000,-14.2000,-14.3000,-14.4000,-14.5000,-14.6000,-14.7000,-14.8000,-14.9000,-15.0000,-15.1000,-15.2000,-15.3000,-15.4000,-15.5000,-15.6000,-15.7000,-15.8000,-15.9000,-16.0000,-16.1000,-16.2000,-16.3000,-16.4000,-16.5000,-16.6000,-16.7000,-16.7000,-16.7000,-16.7000,-16.7000,-16.7000,-16.7000,-16.7000,-16.7000,-16.7000,-16.7000,-16.7000,-16.7000,-16.7000,-16.7000,-16.7000,-16.7000,-16.7000,-16.7000,-16.7000,-16.7000,-16.7000,-16.7000,-16.7000,-16.7000,-16.7000,-16.7000,-16.7000]</rosparam> -->
<rosparam param="map_no">3</rosparam>

<rosparam param="len">[0.848,3.1,15.1,21,21,16.4,2.8,20.3,22.5,1]</rosparam>

<rosparam param="ang">[45,0,-90,0,-180,90,-180,90,-90,-180]</rosparam>

<rosparam param="pose">[0.430, -3.831,0.035]</rosparam>

<rosparam param="obstacle_avoid">true</rosparam>
<!-- if amcl_modify=0,the testpath will not using vo  -->
<rosparam param="amcl_modify">1</rosparam>

<rosparam param="amcl_modify_epoch">1</rosparam>
<!-- modify_range define the maxima range that visual odometry could correct. -->
<rosparam param="modify_range">100</rosparam>
<!-- turning time  -->
<rosparam param="turning_time">9</rosparam>
<!-- moving rate -->
<rosparam param="moving_rate">2</rosparam>

<rosparam param="th_error_limit">1</rosparam>

<rosparam param="path_follow_range">0.3</rosparam>

<rosparam param="covx_limit">0.25</rosparam>

<rosparam param="covy_limit">0.25</rosparam>

<rosparam param="covth_limit">0.1</rosparam>

<param name="Demo_option" value="$(arg option)"/>

<rosparam param="odom_reset_points">15</rosparam>

<rosparam param="odom_belief_points">10</rosparam>

<group ns="$(arg USM_NameSpace)">

    <rosparam param="stop_align_dis">170</rosparam>

    <rosparam param="far_dis">160</rosparam>
    <rosparam param="Lfar_dis">0.5</rosparam>
    <rosparam param="Lclose_dis">0.43</rosparam>
    <rosparam param="Lstop_align_dis">0.8</rosparam>
    <!-- threshold t2 is the number of points ,if it is bigger than t2 , then it is a obstacle. -->
    <rosparam param="close_dis">140</rosparam>
    <!-- threshold t3 is the number of points ,if it is bigger than t2 , then it is a obstacle. -->
    <rosparam param="err">0.01 </rosparam>

    <rosparam param="Kp">0.3 </rosparam>

    <rosparam param="Ki">0.001 </rosparam>
    <rosparam param="md21_max">200 </rosparam>
    <rosparam param="vv">0.1 </rosparam>
</group>

<group ns="$(arg Obstacle_NameSpace)">
  <!-- px1 is the leftmost point's x (175) -->
    <rosparam param="px1">150</rosparam>
    <!-- py1 is the leftmost point's y -->
    <rosparam param="py1">85</rosparam>
    <!-- px is columns of a block (100) -->
    <rosparam param="px">120</rosparam>
    <!-- py us rows of a block -->
    <rosparam param="py">50</rosparam>
    <!-- threshold t1 is the depth value,if it is bigger than t1 , then it is a obstacle point. -->
    <rosparam param="t1">0.01</rosparam>
    <!-- threshold t2 is the number of points ,if it is bigger than t2 , then it is a obstacle. -->
    <rosparam param="t2_c">160</rosparam>
    <!-- threshold t2 is the number of points ,if it is bigger than t2 , then it is a obstacle. -->
    <rosparam param="t2_o">1800</rosparam>

    <rosparam param="t2_oh">3900</rosparam>

    <!-- change the center velocity  -->
    <rosparam param="Vel_c">0.15</rosparam>

    <rosparam param="r1">0.5</rosparam>
    <rosparam param="r2">0.05</rosparam>
    <rosparam param="d2_min">0.2</rosparam>

    <rosparam param="laggy">10</rosparam>
    <rosparam param="moving_lock">True</rosparam>

    <!-- image_show turn on/off -->
    <rosparam param="depth_imgshow">True</rosparam>
    <rosparam param="rgb_show">True</rosparam>
</group>
<!-- launch nodes -->
 <!-- <include file ="$(find tracked_robot)/all_in_one.launch"></include> -->

 <!-- <include file ="$(find tracked_robot)/odom_param.launch"></include>
 <include file ="$(find zed_wrapper)/launch/zed.launch"></include>
 <include file ="$(find my_dynamixel_workbench_tutorial )/launch/position_control.launch"></include>
 <include file ="$(find image_fox)/launch/stair_mark.launch"></include>
 <node pkg="tracked_robot" name="Robot_Control_test" type="Robot_troller" /> -->
 <node pkg="image_fox" type="Demo_stairclimbing" name="Demo_stairclimbing"  output="screen"></node>
  <!-- <node pkg="image_fox" name="showpath" type="showpath" /> -->

</launch>
